app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/ui/fragments/MealListFragment.java

```diff
@@ -38,52 +39,21 @@ public View onCreateView(LayoutInflater inflater,
-                "Couscous aux légumes",
-                "Paella aux crevettes",
-                "Fondue Moitié-Moitié",
-                "Salade légère",
-                "Soupe à l'oignon",
-                "Tarte aux pommes"};
-
-        String[] mealsShortDes = {"Un délicieux poulet au miel",
-                "Un couscous comme à la maison",
-                "Une paella traditionnelle",
-                "Une bonne fondue",
-                "Une salade à la tomate",
-                "De la soupe à l'oignon",
-                "Une bonne tarte aux pommes"};
-
-        String[] mealsLongDes = {"Vous ne pourrez pas résister à ce savoureux poulet",
-                "Ce couscous me fait penser à celui que me faisait mon grand-père",
-                "Recette de paella directement d'Italie !",
-                "blabla fondue",
-                "blabla salade",
-                "blabla soupe",
-                "blabla tarte"};
-
-        ArrayList<Meal> mealArrayList = new ArrayList<>();
-
-        for (int i = 0; i < imageId.length; i++) {
-            Meal meal = new Meal(mealsName[i], mealsShortDes[i], mealsLongDes[i], imageId[i]);
-            mealArrayList.add(meal);
-        }
-
-        MealListAdapter listAdapter = new MealListAdapter(getActivity(), mealArrayList);
-
+        viewModel = new ViewModelProvider(this, new MealListViewModelFactory((NeighborFoodApplication) this.getActivity().getApplication())).get(MealListViewModel.class);
+        MealListAdapter listAdapter = new MealListAdapter(getActivity(), new ArrayList<Meal>());
+        viewModel.getAllMeals().addOnSuccessListener(mealList->{
+            listAdapter.clear();
+            listAdapter.addAll(mealList);
+        }).addOnFailureListener(System.out::println);
         binding.mealListView.setAdapter(listAdapter);
         binding.mealListView.setClickable(true);
-        binding.mealListView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
-            @Override
-            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
-                Intent i = new Intent(getActivity(), MealActivity.class);
-                i.putExtra("name", mealsName[position]);
-                i.putExtra("shortDes", mealsShortDes[position]);
-                i.putExtra("longDes", mealsLongDes[position]);
-                i.putExtra("imageid", imageId[position]);
-                startActivity(i);
-            }
+        binding.mealListView.setOnItemClickListener((parent, view1, position, id) -> {
+            Intent i = new Intent(getActivity(), MealActivity.class);
+            i.putExtra("name",((Meal)(listAdapter.getItem(position))).getName());
+            i.putExtra("shortDes", ((Meal)(listAdapter.getItem(position))).getShortDescription());
+            i.putExtra("longDes", ((Meal)(listAdapter.getItem(position))).getLongDescription());
+            i.putExtra("imageid", ((Meal)(listAdapter.getItem(position))).getImageId());
```

Comment:
here we should remove all the extra in the intent and only leave the MealId so that it can be fetched later from the meal activity